redash/query_runner/elasticsearch.py

```diff
@@ -122,6 +122,8 @@ def _get_query_mappings(self, url):
         for index_name in mappings_data:
             index_mappings = mappings_data[index_name]
             for m in index_mappings.get("mappings", {}):
+                if not hasattr(index_mappings["mappings"][m], "__getitem__"):
```

Comment:
I'm thinking about replacing this block with `try-except`, but this looks like a refactor more than bug fix.